src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -450,6 +471,33 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
     }
   }
 
+  private def isConstrainable(resource: Resource, actions: Set[ResourceAction]): Boolean = {
+    resourceTypes.get(resource.resourceTypeName) match {
+      case Some(resourceType) => resourceType.actionPatterns.exists { actionPattern =>
+        actionPattern.authDomainConstrainable && actions.exists(action => actionPattern.matches(action))
+      }
+      case None =>
+        logger.error(s"Invalid resource type specified. ${resource.resourceTypeName} is not an accepted resource type.")
+        false
+    }
+  }
+
+  private def calculateIntersectionGroup(resource: Resource, policy: AccessPolicy): Future[Set[WorkbenchUserId]] = {
```

Comment:
looking at some other things, I was reminded that this calculation can be done entirely within 1 ldap call. Something like:
`AND filter(EQUALS filter(memberof=policy dn), EQUALS filter(memberof=group1 dn),EQUALS filter(memberof=group2 dn)...)`